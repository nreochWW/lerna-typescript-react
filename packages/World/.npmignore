src/
dist/
legacy/
__tests__/
stories/
tsconfig.json
webpack.config.js