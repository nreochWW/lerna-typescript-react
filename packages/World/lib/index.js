"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
const World = (props) => (React.createElement("div", { style: { border: "1px solid red", marginTop: "10px" } },
    React.createElement("h1", null,
        "Hello World from ",
        props.compiler,
        " and ",
        props.framework,
        "!")));
exports.default = World;
