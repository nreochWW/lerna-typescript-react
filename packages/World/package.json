{
  "name": "world-typescript-react",
  "version": "0.1.0",
  "main": "lib/index.js",
  "license": "MIT",
  "gitHead": "4874514e1688dd45cf9ba8d87ca1c2638126d6c6",
  "scripts": {
    "tsc": "tsc",
    "legacy": "webpack --config legacy/webpack.config.js"
  }
}
