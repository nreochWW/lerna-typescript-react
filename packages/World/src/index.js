"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
const World = (props) => (React.createElement("h1", null,
    "World from ",
    props.compiler,
    " and ",
    props.framework,
    "!"));
exports.default = World;
