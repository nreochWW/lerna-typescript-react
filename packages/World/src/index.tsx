import * as React from "react";
import styled from "styled-components";

export interface WorldProps {
  compiler: string;
  framework: string;
}

const Wrapper = styled.div`
  border: 1px solid red;
  margin-top: 10px;
`;

const World = (props: WorldProps) => (
  <Wrapper>
    <h1>
      Hello World from {props.compiler} and {props.framework}!
    </h1>
  </Wrapper>
);

export default World;
