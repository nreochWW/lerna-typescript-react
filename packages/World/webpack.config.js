const path = require("path");

module.exports = {
  mode: "production",
  entry: {
    world: path.join(__dirname, "legacy", "index.tsx")
  },
  output: {
    filename: "[name].js",
    path: path.join(__dirname, "./dist")
  },
  resolve: {
    extensions: [".ts", ".tsx", ".js"]
  },

  module: {
    rules: [
      {
        test: /\.ts(x?)$/,
        exclude: /node_modules/,
        use: [
          {
            loader: "ts-loader"
          }
        ]
      }
    ]
  }
};
