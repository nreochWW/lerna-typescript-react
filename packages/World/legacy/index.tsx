import * as React from "react";
import * as ReactDOM from "react-dom";

import World from "../src/index";

ReactDOM.render(
  <World compiler="TypeScript" framework="React" />,
  document.getElementById("world")
);
