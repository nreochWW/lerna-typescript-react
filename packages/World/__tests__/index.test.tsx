import * as React from "react";
import { shallow } from "enzyme";
import World from "../src/index";

describe("World package tests", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<World compiler="TypeScript" framework="React" />);
    expect(wrapper).toMatchSnapshot();
  });
});
