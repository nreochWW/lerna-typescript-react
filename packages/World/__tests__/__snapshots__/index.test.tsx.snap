// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`World package tests matches snapshot 1`] = `
<div
  style={
    Object {
      "border": "1px solid red",
      "marginTop": "10px",
    }
  }
>
  <h1>
    Hello World from 
    TypeScript
     and 
    React
    !
  </h1>
</div>
`;
